lib/rubocop/cop/rspec/verified_double_reference.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # Checks for consistent verified double reference style.
+      #
+      # @see https://relishapp.com/rspec/rspec-mocks/docs/verifying-doubles
+      #
+      # This cop can be configured in your configuration using the
+      # `EnforcedStyle` option and supports `--auto-gen-config`.
+      #
+      # @example `EnforcedStyle: const`
+      #   # bad
+      #   let(:foo) do
+      #     instance_double('ClassName', method_name: 'returned_value')
+      #   end
+      #
+      #   # good
+      #   let(:foo) do
+      #     instance_double(ClassName, method_name: 'returned_value')
+      #   end
+      #
+      # @example `EnforcedStyle: string`
+      #   # bad
+      #   let(:foo) do
+      #     instance_double(ClassName, method_name: 'returned_value')
+      #   end
+      #
+      #   # good
+      #   let(:foo) do
+      #     instance_double('ClassName', method_name: 'returned_value')
+      #   end
+      class VerifiedDoubleReference < Base
+        include ConfigurableEnforcedStyle
+
+        MSG = 'Prefer `%<style>s` references for verfied doubles.'
+
+        SUPPORTED_STYLES = %w[const string].freeze
+        RESTRICT_ON_SEND = %i[instance_double instance_spy class_double].freeze
+
+        # @!method verified_double(node)
+        def_node_matcher :verified_double, <<-PATTERN
+          (send nil? {:instance_double :instance_spy :class_double} $_ ...)
+        PATTERN
+
+        def on_send(node)
+          verified_double(node) do |class_reference|
+            return correct_style_detected if preferred_style?(class_reference)
+
+            message = format(MSG, style: style)
+            add_offense(class_reference.loc.expression, message: message) do
+              opposite_style_detected
+            end
+          end
+        end
+
+        private
+
+        def preferred_style?(expectation)
+          map_type(expectation.type).equal?(style)
+        end
+
+        def map_type(type)
+          return :string if type == :str
+
+          type
+        end
```

Comment:
Did you consider having an actual map (Hash instance) instead of a `#map_type` method?

Reply:
I did consider it. Moved to a map now that I'm using `constant` as the style name